src/flamenco/vm/fd_vm_interp_core.c

```diff
@@ -365,7 +365,7 @@
     ulong vaddr           = reg_dst + (ulong)(long)offset;
     ulong haddr           = fd_vm_mem_haddr( vm, vaddr, sizeof(uint), region_haddr, region_st_sz, 1, 0UL, &is_multi_region );
     int   sigsegv         = !haddr;
-    if( FD_UNLIKELY( sigsegv ) ) goto sigsegv; /* Note: untaken branches don't consume BTB */ /* FIXME: sigbus */
+    if( FD_UNLIKELY( sigsegv ) ) { vm->segv_store_vaddr = vaddr; goto sigsegv; } /* Note: untaken branches don't consume BTB */ /* FIXME: sigbus */
```

Comment:
We only need to write this value in this case right? https://github.com/firedancer-io/firedancer/blob/main/src/flamenco/vm/fd_vm_private.h#L322

It would be great to minimise the amount of checks/writes we do in the vm hot path.